### PR TITLE
CAMEL-14425: Polling Optimization: Consider number of maximum message…

### PR DESCRIPTION
**Pull request by mash-sap**
…s to process earlier during the poll to prevent running through multiple for-loops with all messages

### CODE REVIEW

File: components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java [hunk 1]
```diff
@@ -95,6 +95,30 @@ protected void doStop() throws Exception {
         super.doStop();
     }
 
+    /**
+     * Returns the max number of messages to be processed. Will return -1 if no maximum is set
+     *
+     * @return
```
Comment (oscerd) [inline components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java]: please remove this

Comment (mash-sap) [reply components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java]: Comment removed

File: components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java [hunk 2]
```diff
@@ -95,6 +95,25 @@ protected void doStop() throws Exception {
         super.doStop();
     }
```
Comment (oscerd) [inline components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java]: I meant only the '@return' part :-)

Comment (mash-sap) [reply components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java]: Ok. I understood to put the comment in the adoc instead. I updated the commit. Please check :)

Comment (oscerd) [reply components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java]: My fault. It looks good. When you can please a note about the change in the adoc, not this comment :-)

Comment (mash-sap) [reply components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java]: Sorry, I'm not familiar with the adoc. Do I need a editor for it? Where should I document my change? I could not see a suitable place in the mail-component.adoc.

Comment (oscerd) [reply components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java]: You just need to add a section here https://github.com/apache/camel/blob/master/components/camel-mail/src/main/docs/mail-component.adoc

By the way, no problem, we can do this later.

Comment (mash-sap) [reply components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java]: Sorry I still don't get it :) What exactly you like me to document? The commit is just code optimization, right? :)

Comment (oscerd) [reply components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java]: Usually people notice performance optmizations, but sometime for special cases people may also note problems, so we need to add a note about what you've done between 3.0 and 3.1

Comment (mash-sap) [reply components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java]: I updated the mail.component.adoc. Please have a look.

**Pull request merged**